lib/game.ex

```diff
@@ -124,4 +128,128 @@ defmodule Game do
   def get_marker(game), do: game.player_2_mark
 
   def update_board(board, game), do: %{game | board: board}
+
+  def next_open(game), do: List.first(get_available_moves(game))
+
+  def handle_next_move(game) do
+    move = next_open(game)
+    player_move_list = determine_move_list(game)
+    game = update_available_moves(move, game)
+    game = update_player_move_list(game, player_move_list, move)
+    place_marker(move, game)
+  end
+
+  def best_move(game) do
+    # best_score = -100
+    # new_move = nil
+    game = assign_maximizer(game)
+
+    # Iterate through every cell
+    score_move_list =
+      Enum.map([1, 2, 3, 4, 5, 6, 7, 8, 9], fn move ->
+        IO.puts("\r\n\r\nTrying == #{move} ")
+
+        # Grab that move number in case we want to use it
+        IO.puts("AVAIL MOVES BELOW")
+        IO.inspect(game.available_moves)
+        IO.puts("AVAIL MOVES ABOVE")
+
+        IO.puts("MOVE IN AVAIL MOVES?")
+        IO.inspect(move in game.available_moves)
+        IO.puts("MOVE IN AVAIL MOVES?")
+
+        # If move is in available moves continue
+        if move in game.available_moves do
+          # IO.puts("BEST SCORE BEFORE ==== #{best_score}")
+          IO.puts("DOING == #{move} ")
+          # play the next available move
+          game = handle_move(move, game)
+          game = check_for_win(game)
+          {move, score(game)}
+
+        else
+          # If move is not in available moves skip
+          IO.puts("SKIPPING == #{move}\n")
+        end
+      end)
+
+    move_list = Enum.filter(score_move_list, fn elem -> elem != :ok end)
```

Comment:
i'm a little lost, but does this ever return `false` for any element in your `move_list`? `move_list` is a list of `{move, score}` tuples || `nil`, right?

it also feels a little odd in this context to be using this idiom to specifically filter on things that are _not_ `:ok` when we're on the happy/success path.

Reply:
This is a syntax issue. If I could just not return the :ok s it would be a bit better and no filter would be needed. 

At the end of this I started thinking that I could store the best_move and best_score in the game object itself. Once I do that I may be able to just let this game object fall through until all combos are done, then just grab the best_move key/ value in the object. 

@new_game %{
    :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
    :win? => false,
    :winner => nil,
    :player_1_moves => [],
    :player_2_moves => [],
    :player_1_mark => "X",
    :player_2_mark => "O",
    :current_player => 1,
    :player_1_type => :human,
    :player_2_type => :human,
    :maximizer => nil,
    :best_score_move => %{
      :best_score => -100,
      :best_move => nil
    },
    :board => %{
      1 => "1",
      2 => "2",
      3 => "3",
      4 => "4",
      5 => "5",
      6 => "6",
      7 => "7",
      8 => "8",
      9 => "9"
    }
  }